src/lld/moviebooking/class_diagram.puml

```diff
@@ -1,3 +1,323 @@
 @startuml
+'Main Entity and Relational mapping entity
 
+package com.movie_booking.entity.main{
+    class Identifier{
+        - id: long
+        - name: String
+    }
+    class User extends Identifier{
+        - email: String
+        - phoneNo: String
+    }
+    class Movie extends Identifier{
+        - crew: List<Crew>
+        - supportedLanguages: List<LanguageCode>
+    }
+    class Crew extends Identifier{
+        - crewType: CrewType
+    }
+    enum CrewType{
+        Cameraman
```

Comment:
since enum is named after crew, it should be valued as camera, design as a general group rather than pointing to a particular person like cameraman,

Also how does Hero, Heroine be a crewType. They may be said as Casts or CrewMemberType

Reply:
Changed it to hold both enum in single enum to scale better and identify cast or crew in a single operation
`    class CrewCast extends Identifier{
        - crewCastType: CrewCastType
    }
    enum CrewCastType{
        Cameraman(MovieMemberType.Crew)
        Designer(MovieMemberType.Crew)
        Director(MovieMemberType.Crew)
        Singer(MovieMemberType.Crew)
        Hero(MovieMemberType.Cast)
        Heroine(MovieMemberType.Cast)
        - <<final>> movieMemberType: MovieMemberType
        + CrewCastType(movieMemberType: movieMemberType)
    }`
    enum MovieMemberType{
         Crew,
         Cast
    }